torchmetrics/functional/classification/auroc.py

```diff
@@ -86,7 +86,7 @@ def _auroc_compute(
             tpr = [o[1] for o in output]
```

Comment:
Then we should probably have another else here
```suggestion
            tpr = [o[1] for o in output]
        else:
            raise ValueError('Detected input to be `multilabel` but you did not provide `num_classes` argument')
```